aerogear/mobile-security-service
# feat(ui): Add navigation modal to UI (AEROGEAR-8473)

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8473

## What
Create a navigation modal

## Why
To advise the end user they have unsaved changes before leaving the detailed UI page.

## Verification Steps
1. Checkout this branch locally
2. Modify the isAppDetailedDirty property in the reducer initalState to true
3. Run the server and UI using `make serve`
4. Navigate to the detailed app screen
5. Click the back button in the browser or the header
6. Verify the navigation modal appears
7. Verify the `X` and `Stay` buttons close the modal and keep you on the same screen
8. Verify the `Leave` button navigates back to the landing page

## Checklist:


## Progress
